src/analysis/mesh/qgsmeshtriangulation.cpp

```diff
@@ -0,0 +1,339 @@
+  if ( valueAttribute >= 0 )
+    value = feature.attribute( valueAttribute ).toDouble();
+
+  while ( vit != geom.vertices_end() )
+  {
+    if ( feedback && feedback->isCanceled() )
+      break;
+    if ( valueAttribute < 0 )
+      mTriangulation->addPoint( *vit );
+    else
+    {
+      mTriangulation->addPoint( QgsPoint( QgsWkbTypes::PointZ, ( *vit ).x(), ( *vit ).y(), value ) );
+    }
+    ++vit;
+  }
+}
+
+void QgsMeshTriangulation::addBreakLinesFromFeature( const QgsFeature &feature, int valueAttribute, const QgsCoordinateTransform &transform, QgsFeedback *feedback )
+{
+  double valueOnVertex = 0;
+  if ( valueAttribute >= 0 )
+    valueOnVertex = feature.attribute( valueAttribute ).toDouble();
+
+  if ( feedback )
+    feedback->setProgress( 0 );
+
+  //from QgsTinInterpolator::insertData()
+  std::vector<const QgsCurve *> curves;
+  QgsGeometry geom = feature.geometry();
+  try
+  {
+    geom.transform( transform, QgsCoordinateTransform::ForwardTransform, true );
+  }
+  catch ( QgsCsException &cse )
+  {
+    Q_UNUSED( cse )
+    QgsDebugMsgLevel( QStringLiteral( "Caught CRS exception %1" ).arg( cse.what() ), 4 );
+    return;
+  }
+
+  if ( QgsWkbTypes::geometryType( geom.wkbType() ) == QgsWkbTypes::PolygonGeometry )
+  {
+    std::vector< const QgsCurvePolygon * > polygons;
+    if ( geom.isMultipart() )
+    {
+      const QgsMultiSurface *ms = qgsgeometry_cast< const QgsMultiSurface * >( geom.constGet() );
+      for ( int i = 0; i < ms->numGeometries(); ++i )
+      {
+        polygons.emplace_back( qgsgeometry_cast< const QgsCurvePolygon * >( ms->geometryN( i ) ) );
+      }
+    }
+    else
+    {
+      polygons.emplace_back( qgsgeometry_cast< const QgsCurvePolygon * >( geom.constGet() ) );
+    }
+
+    for ( const QgsCurvePolygon *polygon : polygons )
+    {
+      if ( feedback && feedback->isCanceled() )
+        break;
+      if ( !polygon )
+        continue;
+
+      if ( polygon->exteriorRing() )
+        curves.emplace_back( polygon->exteriorRing() );
+
+      for ( int i = 0; i < polygon->numInteriorRings(); ++i )
+      {
+        if ( feedback && feedback->isCanceled() )
+          break;
+        curves.emplace_back( polygon->interiorRing( i ) );
+      }
+    }
+  }
+  else
+  {
+    if ( geom.isMultipart() )
+    {
+      const QgsMultiCurve *mc = qgsgeometry_cast< const QgsMultiCurve * >( geom.constGet() );
+      for ( int i = 0; i < mc->numGeometries(); ++i )
+      {
+        if ( feedback && feedback->isCanceled() )
+          break;
+        curves.emplace_back( qgsgeometry_cast< const QgsCurve * >( mc->geometryN( i ) ) );
+      }
+    }
+    else
+    {
+      curves.emplace_back( qgsgeometry_cast< const QgsCurve * >( geom.constGet() ) );
+    }
+  }
+
+  int i = 0;
+  for ( const QgsCurve *curve : curves )
+  {
+    if ( !curve )
+      continue;
+
+    if ( feedback )
+    {
+      if ( feedback->isCanceled() )
+        break;
+
+      feedback->setProgress( 100 * i / curves.size() );
```

Comment:
I'd remove this -- otherwise the progress being reported here is just the progress for this one feature, and it'll cause the progress reports for the caller to jump all over the place.